settings/resource_default_namespace_settings.go

```diff
@@ -0,0 +1,178 @@
+package settings
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// NewDefaultNamespaceSettingsAPI creates a DefaultNamespaceSettingsAPI instance
+func NewDefaultNamespaceSettingsAPI(ctx context.Context, m any) DefaultNamespaceSettingsAPI {
+	client := m.(*common.DatabricksClient)
+	return DefaultNamespaceSettingsAPI{client, ctx}
+}
+
+// DefaultNamespaceSettingsAPI exposes the Default Namespace Settings API
+type DefaultNamespaceSettingsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func (a DefaultNamespaceSettingsAPI) isEtagVersionError(err error) bool {
+	var aerr *apierr.APIError
+	if !errors.As(err, &aerr) {
+		return false
+	}
+	return aerr.StatusCode == http.StatusNotFound || (aerr.StatusCode == http.StatusConflict && aerr.ErrorCode == "RESOURCE_CONFLICT")
+}
+
+func (a DefaultNamespaceSettingsAPI) getEtagFromError(err error) (string, error) {
+	if !a.isEtagVersionError(err) {
+		return "", err
+	}
+	errorInfos := apierr.GetErrorInfo(err)
+	if len(errorInfos) > 0 {
+		metadata := errorInfos[0].Metadata
+		if etag, ok := metadata["etag"]; ok {
+			return etag, nil
+		}
+	}
+	return "", fmt.Errorf("error fetching the default workspace namespace settings: %w", err)
+}
+
+func (a DefaultNamespaceSettingsAPI) DeleteWithRetry(etag string) (string, error) {
```

Comment:
should we use: https://github.com/databricks/databricks-sdk-go/blob/main/retries/retries.go?